sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -607,14 +632,60 @@ extension StorageBlobClient {
         manager.resumeAll()
     }
 
-    /// Retrieve a single Transfer object by its id.
+    /// Retrieve a single `Transfer` object by its id.
     ///
     /// - Parameters:
     ///   - id: The id of the transfer to retrieve.
     public func transfer(withId id: UUID) -> Transfer? {
         return manager.transfer(withId: id)
     }
 
+    /// Retrieve a single `BlobTransfer` object by name and container.
+    ///
+    /// - Parameters:
+    ///   - blob: The name of the blob.
+    ///   - container: The name of the container.
+    ///   - type: The `TransferType` to retrieve. Leave blank to retrieve either uploads or downloads.
+    /// - Returns: The matching `BlobTransfer` if found, or nil.
+    public func transfer(
+        forBlob blob: String,
+        inContainer container: String,
+        ofType type: TransferType? = nil
+    ) -> BlobTransfer? {
```

Comment:
Definitely need some way to match a transfer to something that is a property of the blob on the service. The UUID used for transfer ID is completely random and doesn't serve this purpose. The closest thing to a key for a blob that I'm aware of is its full URL.

Reply:
Updated to use transfer collection extensions.